Modules/timemodule.c

```diff
@@ -834,101 +824,127 @@ time_strftime(PyObject *module, PyObject *args)
     else if (buf.tm_isdst > 1)
         buf.tm_isdst = 1;
 
-#ifdef HAVE_WCSFTIME
-    format = PyUnicode_AsWideCharString(format_arg, NULL);
-    if (format == NULL)
-        return NULL;
-    fmt = format;
-#else
-    /* Convert the unicode string to an ascii one */
-    format = PyUnicode_EncodeLocale(format_arg, "surrogateescape");
-    if (format == NULL)
+    format_size = PyUnicode_GET_LENGTH(format_arg);
+    format = PyMem_Malloc((format_size + 1)*sizeof(time_char));
```

Comment:
I suggest to add an explicit integer overflow check: fail with PyErr_NoMemory() on overflow.